mono/utils/mono-threads-state-machine.c

```diff
@@ -170,12 +179,15 @@ This transition initiates the suspension of another thread.
 
 Returns one of the following values:
 
-- AsyncSuspendInitSuspend: Thread suspend requested, async suspend needs to be done.
-- AsyncSuspendAlreadySuspended: Thread already suspended, nothing to do.
-- AsyncSuspendBlocking: Thread in blocking state
+- ReqSuspendInitSuspendRunning: Thread suspend requested, caller must initiate suspend.
+- ReqSuspendInitSuspendBlocking: Thread in blocking state, caller may initiate suspend.
+- ReqSuspendAlreadySuspended: Thread was already suspended and not executing, nothing to do.
+- ReqSuspendAlreadySuspendedBlocking: Thread was already in blocking and a suspend was requested
+                                      and the thread is still executing (perhaps in a syscall),
+                                      nothing to do.
 */
-MonoRequestAsyncSuspendResult
-mono_threads_transition_request_async_suspension (MonoThreadInfo *info)
+MonoRequestSuspendResult
+mono_threads_transition_request_initiate_suspension (MonoThreadInfo *info)
```

Comment:
`mono_threads_transition_request_suspension` should be enough and provides a good counter-part to `mono_threads_transition_request_resume`.